### PR TITLE
Reviewing the documentation for local development

### PR DESCRIPTION
**Pull request by agoncal**

### CODE REVIEW

File: docs/assets/architecture.png
Comment (sinedied) [inline docs/assets/architecture.png:1]: Please don't change it directly!
- The source is: architecture.excalidraw
- Other workshop versions use this very same image (like Node.js)

OpenAI is not supported (and probably won't since we'll be enforcing managed identity and move away from API keys...), maybe it's a better idea to just rename "Azure OpenAI" to something more generic like "AI models"?

File: docs/sections/01-intro.md
Comment (sinedied) [inline docs/sections/01-intro.md:20]:
```suggestion
Our application consists of five main components:
```

File: docs/sections/10-deployment.md
Comment (sinedied) [inline docs/sections/10-deployment.md:7]:
```suggestion
Run this command from the root of the project to build and deploy the application (this command deploys all services listed in the `azure.yaml` file located in the project root):
```

File: docs/sections/ai-search/10-deployment.md
Comment (sinedied) [inline docs/sections/ai-search/10-deployment.md:7]:
```suggestion
Run this command from the root of the project to build and deploy the application (this command deploys all services listed in the `azure.yaml` file located in the project root):
```

File: docs/sections/java-quarkus/08-website.md
Comment (sinedied) [inline docs/sections/java-quarkus/08-website.md:1]: We already discussed this, but the whole section is optional.

The problem I see (and already experienced) with adding `(Optional)` statement in the individual sections is that it's confusing as hell: the individual section are not to be cherry picked, it's either do all or don't but adding optional like this is even less clear than it was.

The intent of the skip sections was to make the workshop more modular and gain time, and the whole workflow is "not interested in **XXX**? run the curl command and move on to the next".

Any other use case is at the discretion of the user. This is currently the only skip section here, but more are to be added. Look at the JS version for example: https://moaw.dev/workshop/gh:sinedied/openai-rag/docs/workshop-qdrant.md?step=8
Every skip section is meant to work the same way, I don't see why this one should be different?

**Review by sinedied — CHANGES_REQUESTED** [review]
Sent some suggestions.

Please don't merge until the java code for ollama is implemented and working, this gets published directly once merged!

File: docs/sections/java-quarkus/08-website.md
Comment (agoncal) [reply docs/sections/java-quarkus/08-website.md:1]: This is the Java version of the workshop. Having to write front-end code is really not needed (not talking about installing NPM and so on). So what do you propose ? If we skip the entire section there is no way we will have the front-and running locally. That's why I added Quinoa in the very first place. Java developers don't care about developping the UI. Quarkus + Quinoa hides everything. So I think we should just give the UI in this workshop and build it with Quinoa.

File: docs/sections/01-intro.md
Comment (sinedied) [inline docs/sections/01-intro.md:1]: note to myself: also update other variants with this :)

File: docs/sections/10-deployment.md
Comment (sinedied) [inline docs/sections/10-deployment.md:1]: note for me: update also in other variants

**Comment by agoncal** [comment]
Fix #26

**Review by sinedied — APPROVED** [review]

**Pull request merged**
